frontend/src/entry/model/ModelImageList.tsx

```diff
@@ -22,12 +22,26 @@ type PartialModelImageListProps =
 
 type ModelImageListProps = {
   model: EntryInterface
+  setRegistryError?: (value: boolean) => void
   readOnly?: boolean
 } & PartialModelImageListProps
 
-export default function ModelImageList({ model, value, onChange, readOnly = false, multiple }: ModelImageListProps) {
+export default function ModelImageList({
+  model,
+  value,
+  onChange,
+  setRegistryError,
+  readOnly = false,
+  multiple,
+}: ModelImageListProps) {
   const { modelImages, isModelImagesLoading, isModelImagesError } = useGetModelImages(model.id)
 
+  useEffect(() => {
+    if (setRegistryError) {
+      setRegistryError(!!isModelImagesError)
+    }
+  }, [isModelImagesError, setRegistryError])
```

Comment:
As a learning point, be aware that it can be dangerous to put function props in any dependency arrays (`setRegistryError` in this case). JS functions are recreated on every render and will therefore always trigger hooks that depend on it unless you wrap it with a `useCallback` that itself doesn't rely on an unstable function.

I've already added comments for where those `useCallbacks` are needed in this PR, but thought it worth putting this here so that you can get the full picture.